src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Collect.cs

```diff
@@ -0,0 +1,472 @@
+            }
+        }
+
+        // This method returns true if there are special suggestions for a particular parameter of the function.
+        public override bool HasSuggestionsForParam(int argumentIndex)
+        {
+            Contracts.Assert(argumentIndex >= 0);
+
+            return argumentIndex == 0;
+        }
+
+        public override bool TryGetDataSourceNodes(PowerFx.Syntax.CallNode callNode, TexlBinding binding, out IList<FirstNameNode> dsNodes)
+        {
+            Contracts.AssertValue(callNode);
+            Contracts.AssertValue(binding);
+
+            dsNodes = new List<FirstNameNode>();
+            if (callNode.Args.Count != 2)
+            {
+                return false;
+            }
+
+            var args = Contracts.VerifyValue(callNode.Args.Children);
+            var arg1 = Contracts.VerifyValue(args[1]);
+
+            // Only the second arg can contribute to the output for the purpose of delegation
+            return ArgValidators.DataSourceArgNodeValidator.TryGetValidValue(arg1, binding, out dsNodes);
+        }
+
+        public override IEnumerable<Identifier> GetIdentifierOfModifiedValue(TexlNode[] args, out TexlNode identifierNode)
+        {
+            Contracts.AssertValue(args);
+
+            identifierNode = null;
+            if (args.Length == 0)
+            {
+                return null;
+            }
+
+            var firstNameNode = args[0]?.AsFirstName();
+            identifierNode = firstNameNode;
+
+            if (firstNameNode == null)
+            {
+                return null;
+            }
+
+            var identifiers = new List<Identifier>
+            {
+                firstNameNode.Ident
+            };
+            return identifiers;
+        }
+
+        public override bool IsAsyncInvocation(PowerFx.Syntax.CallNode callNode, TexlBinding binding)
+        {
+            Contracts.AssertValue(callNode);
+            Contracts.AssertValue(binding);
+
+            return Arg0RequiresAsync(callNode, binding);
+        }
+
+        public static DType GetCollectedTypeForGivenArgType(Features features, DType argType, TexlNode arg, ref Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            var singleColumnRecordType = GetCollectedTypeForGivenArgType(features, argType);
+
+            if (!argType.IsPrimitive)
+            {
+                return argType;
+            }
+
+            //CollectionUtils.Add(ref nodeToCoercedTypeMap, arg, singleColumnRecordType);
+
+            return singleColumnRecordType;
+        }
+
+        public static DType GetCollectedTypeForGivenArgType(Features features, DType argType)
+        {
+            Contracts.Assert(argType.IsValid);
+
+            if (!argType.IsPrimitive)
+            {
+                return argType;
+            }
+
+            // Passed a scalar; make a record out of it, using a name that depends on the type.
```

Comment:
>, using a name that depends on the type. [](http://example.com/codeflow?start=55&length=40)

This part of the comment can be changed, since the name is only 'Value' now (wrapped record is {Value:<primitive>})